### PR TITLE
Tel 107 add unit conversion to pis transformer

### PR DESCRIPTION
**Pull request by ideen1**

### CODE REVIEW

File: client/src/components/transformers/PISTransformer.tsx [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
 
 type RangeCheckedFieldDataProps = {
   fieldData: I_PISFieldData;
 };
 
+function FieldUnitsHandler(
+  unit: UnitType | undefined,
+  value: string | number | boolean,
+) {
+  const appState = useAppState();
```
Comment (brian-ngyn) [inline client/src/components/transformers/PISTransformer.tsx:17]: you should destructure `appState` so you don't have to do `appState.currentAppState` everytime

**Review by brian-ngyn — CHANGES_REQUESTED** [review]

File: client/src/components/transformers/PISTransformer.tsx [hunk 2]
```diff
@@ -1,10 +1,73 @@
...
+
+  let unitReturn: string | undefined;
+  let valueReturn: string | number | boolean;
+  unitReturn = unit;
+  valueReturn = value;
+
+  if (unit === UnitType.TEMP) {
```
Comment (brian-ngyn) [inline client/src/components/transformers/PISTransformer.tsx:24]: can we use a case switch statement?

```
switch (unit) {
  case UnitType.TEMP:
    // your code here
    break;
  
  case UnitType.SPEED:
    // your code here
    break;
  
  case UnitType.DISTANCE:
    // your code here
    break;

  default:
    unitReturn = "";
    valueReturn = value;
}
```

**Review by brian-ngyn — CHANGES_REQUESTED** [review]

File: client/src/contexts/AppStateContext.tsx [hunk 3]
```diff
@@ -99,3 +99,5 @@ export function AppStateContextProvider({ children }: Props) {
 export function useAppState(): IAppStateReturn {
   return useContext(appStateContext);
 }
+
+export default APPUNITS;
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:103]: Remove this

**Comment by ideen1** [comment]
Looks super good. Just one last thing: Can you round the miles per hour to 0 decimal places? Right now it shows 20 decimal places.

**Review by justin-phxm — APPROVED** [review]
LGTM

**Review by brian-ngyn — APPROVED** [review]

**Pull request merged**